src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -262,7 +262,7 @@ protected override ImmutableArray<PendingBranch> RemoveReturns()
                 var foundAwait = result.Any(pending => pending.Branch?.Kind == BoundKind.AwaitExpression);
                 if (!foundAwait)
                 {
-                    Diagnostics.Add(ErrorCode.WRN_AsyncLacksAwaits, currentMethodOrLambda.Locations[0]);
+                    Diagnostics.Add(ErrorCode.WRN_AsyncLacksAwaits, currentMethodOrLambda.NarrowLocation());
```

Comment:
Ok, that turned out to be painful. `TestDiagnosticAnalyzerDriver.GetDiagnosticsAsync` checks whether the span from the diagnostics intersect with the currently selected span...
I'll keep the diagnostic for `WRN_AsyncLacksAwaits` as before...